src/handlers/logbook/addDetail.js

```diff
@@ -0,0 +1,51 @@
+import db from "../../db";
+import log from "../../utils/log";
+import HttpError from "../../exceptions/HttpError";
+import HttpStatusCode from "../../utils/HttpStatusCode";
+
+export default async (req, res, next) => {
+  const { logbookId } = req.params;
+
+  let logbook;
+  try {
+    logbook = await db.Logbook.findOne({
+      where: { id: logbookId },
+    });
+  } catch (err) {
+    log.error(`Error finding logbook ${logbookId}, ${err.message}`);
+    return next(
+      new HttpError(
+        HttpStatusCode.INTERNAL_SERVER_ERROR,
+        "Internal Server Error"
+      )
+    );
+  }
+
+  if (logbook === null) {
+    return next(new HttpError(HttpStatusCode.NOT_FOUND, "Not Found"));
+  }
+
+  if (logbook.user_id !== req.auth.id) {
+    return next(new HttpError(HttpStatusCode.FORBIDDEN, "Forbidden"));
+  }
```

Comment:
Feel like I'm seeing some stuff that could be DRY'd

This snippet in particular.

Reply:
Perhaps a getLogbook function that returns a logbook or throws an error that can be caught?